x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -339,14 +339,21 @@ public void onIndexModule(IndexModule indexModule) {
     }
 
     public List<RestHandler> getRestHandlers(
-        Settings settings,
+        Settings nodeSettings,
         RestController restController,
         ClusterSettings clusterSettings,
         IndexScopedSettings indexScopedSettings,
         SettingsFilter settingsFilter,
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<DiscoveryNodes> nodesInCluster
     ) {
+        indexScopedSettings.addSettingsUpdateConsumer(s -> {}, List.of(IndexMetadata.INDEX_BLOCKS_WRITE_SETTING), settings -> {
```

Comment:
I'm surprised that the setting update consumer registration happens here in the `getRestHandlers()`. I'd expect it to be registered using `onIndexModule(IndexModule)` so that the consumer is registered once and only for s/s shards. Does that make a difference?

Reply:
I started off that way as well, but then noticed that `onIndexModule` has the wrong lifecycle. It gets called when a shard of a given index is allocated on a node, or when a temporary IndexService is created on the master (or mappings updated on the master), but might not generally called for every existing index on the master. `MetadataUpdateSettingsService` does not instantiate `IndexService`, and therefore does not call `onIndexModule`. I'm interested in registering with the global `IndexScopedSettings` object that is referenced by `MetadataUpdateSettingsService` on the master. There are currently not other plugin methods that expose this except `getRestHandlers`. Given that we already implement that method, I did not see the need to make an infrastructural change to e.g. add `IndexScopedSettings` to `createComponents` or any other plugin extension method.

Reply:
I originally thought this would not be an option, but it indeed is. Thank you for the feedback, I've reworked the PR now according to your suggestion.